src/components/button/createButton.ts

```diff
@@ -10,21 +10,29 @@ export interface ButtonState extends WidgetState, FormFieldMixinState<string> {
 
 export interface ButtonProperties extends WidgetProperties {
 	label?: string;
+	onClick?(event: MouseEvent): void;
 }
 
 export interface ButtonOptions extends WidgetOptions<ButtonState, ButtonProperties>, FormFieldMixinOptions<any, ButtonState> { }
 
-export type Button = Widget<ButtonState, ButtonProperties> & FormFieldMixin<string, ButtonState>;
+export type Button = Widget<ButtonState, ButtonProperties> & FormFieldMixin<string, ButtonState> & {
+	onClick(event?: MouseEvent): void;
+};
 
 export interface ButtonFactory extends ComposeFactory<Button, ButtonOptions> { }
 
 const createButton: ButtonFactory = createWidgetBase
 	.mixin(createFormFieldMixin)
 	.mixin({
 		mixin: {
+			onClick(this: Button, event: MouseEvent) {
+				if (this.properties.onClick) {
+					this.properties.onClick.call(this, event);
```

Comment:
I don't think this is meant to be called on `this` scope, rather should be just whatever the function passed in properties is bound to. So that makes this `onclick` slightly less ugly I think:
```ts
onClick(this: Button, event: MouseEvent) {
	this.properties.onclick && this.properties.onclick(event);
}
```